pulp_container/tests/functional/api/test_push_signatures.py

```diff
@@ -0,0 +1,155 @@
+"""Tests that verify that an image signature can be pushed to Pulp."""
+import gnupg
+import base64
+import json
+import requests
+import shutil
+import subprocess
+import pytest
+import tempfile
+
+from pulp_container.tests.functional.constants import REGISTRY_V2_REPO_PULP
+from pulp_container.constants import SIGNATURE_TYPE
+
+KEY = "https://raw.githubusercontent.com/pulp/pulp-fixtures/master/common/GPG-PRIVATE-KEY-pulp-qe"
+
+
+@pytest.fixture(scope="session")
+def tempdir():
+    """A temporary directory."""
+    directory = tempfile.mkdtemp()
+    yield directory
+    shutil.rmtree(directory)
```

Comment:
```suggestion
    with tempfile.TemporaryDirectory() as directory:
        yield directory
```
Context managers are _made_ for this...